schematic/schemas/json_schema_generator.py

```diff
@@ -0,0 +1,747 @@
+"Data Model Json Schema"
+
+import logging
+import os
+from typing import Union, Any
+from dataclasses import dataclass, field, asdict
+import networkx as nx  # type: ignore
+
+from schematic.schemas.data_model_graph import DataModelGraphExplorer
+from schematic.schemas.data_model_relationships import DataModelRelationships
+from schematic.utils.schema_utils import get_json_schema_log_file_path
+from schematic.utils.validate_utils import rule_in_rule_list
+
+from schematic.utils.io_utils import export_json
+
+
+logger = logging.getLogger(__name__)
+
+# A dict where the keys are type validation rules, and the values are their JSON Schema equivalent
+TYPE_RULES = {
+    "str": "string",
+    "num": "number",
+    "float": "number",
+    "int": "integer",
+    "bool": "boolean",
+}
+
+
+@dataclass
+class PropertyData:
+    """A Dataclass representing data about a JSON Schema property from its validation rules"""
+
+    property_type: Union[str, None] = None
+    is_array: bool = False
+    range_min: Union[float, None] = None
+    range_max: Union[float, None] = None
+
+
+@dataclass
+class JSONSchema:  # pylint: disable=too-many-instance-attributes
+    """A dataclass representing a JSON Schema"""
+
+    schema_id: str = ""
+    title: str = ""
+    schema: str = "http://json-schema.org/draft-07/schema#"
+    type: str = "object"
+    description: str = "TBD"
+    properties: dict[str, Any] = field(default_factory=dict)
+    required: list[str] = field(default_factory=list)
+    all_of: list[dict[str, Any]] = field(default_factory=list)
```

Comment:
Can you add a docstring to the class (or the fields themselves) that describes the purpose of each attribute?